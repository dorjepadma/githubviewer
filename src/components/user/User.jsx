import React from 'react';
import PropTypes from 'prop-types';

const User = ({ image, username, followers, following, profileUrl, }) => (

  <figure style={{ margin: '1em' }}>
    <section>
      <img alt={username} style={{ width: '25%' }} src={image} />
      <div style={{ fontWeight: 'bold' }}>
        <h3>Name: {username} </h3>
        <p>Followers: {followers} </p>
        <p>Following: {following} </p>
        <a href={profileUrl} target="_blank" rel="noopener noreferrer">Profile</a>
      </div>
    </section>
  </figure>
   
);

User.propTypes = {
  image: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  profileUrl: PropTypes.string.isRequired
};
export default User;
