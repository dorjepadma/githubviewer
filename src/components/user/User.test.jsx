import React from 'react';
import { shallow } from 'enzyme';
import User from './User';

describe('User', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<User name='dorjepadma'
      followers={1}
      following={1}
      profileUrl='https://api.github.com/users/dorjepadma'
      avatar_url='https://avatars0.githubusercontent.com/u/59029366?v=4' />);
    expect(wrapper).toMatchSnapshot();
  });
});
