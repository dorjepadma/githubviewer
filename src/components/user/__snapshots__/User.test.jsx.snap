// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User matches a snapshot 1`] = `
<figure>
  <img
    alt="dorjepadma"
    src="https://avatars0.githubusercontent.com/u/59029366?v=4"
  />
  <figcaption>
    <p>
      dorjepadma
    </p>
    <p>
      1
    </p>
    <p>
      1
       
    </p>
    <p />
  </figcaption>
</figure>
`;
